prov/psm3/Makefile.include

```diff
@@ -29,7 +29,7 @@ _psm3_cppflags = \
 chksum_srcs = $(_psm3_files)
 
 if HAVE_PSM3_SRC
-_psm3_cflags = -mavx2
+_psm3_cflags =
```

Comment:
unrelated to this PR, but why is there a check for HAVE_PSM3_SRC?  The provider is the psm3 source, correct?

Reply:
For Internal testing we sometimes build as separate lib like psm2.